extensions/hibernate-orm/deployment/src/main/java/io/quarkus/hibernate/orm/deployment/HibernateOrmProcessor.java

```diff
@@ -806,10 +807,12 @@ private static void producePersistenceUnitDescriptorFromConfig(
         }
 
         // Query
-        if (persistenceUnitConfig.fetch.batchSize > 0) {
-            setBatchFetchSize(descriptor, persistenceUnitConfig.fetch.batchSize);
-        } else if (persistenceUnitConfig.batchFetchSize > 0) {
-            setBatchFetchSize(descriptor, persistenceUnitConfig.batchFetchSize);
+        int batchSize = firstPresent(persistenceUnitConfig.fetch.batchSize, persistenceUnitConfig.batchFetchSize)
+                .orElse(16);
```

Comment:
I think we could simplify things and drop the deprecated one for 2.0. It has been deprecated since 09/2020 so quite a few versions.

Not blocking this PR though.